browser/brave_wallet/brave_wallet_provider_delegate_impl.cc

```diff
@@ -86,4 +93,21 @@ void BraveWalletProviderDelegateImpl::RequestEthereumPermissions(
       content::RenderFrameHost::FromID(routing_id_), std::move(callback)));
 }
 
+void BraveWalletProviderDelegateImpl::GetAllowedAccounts(
+    GetAllowedAccountsCallback callback) {
+  EnsureConnected();
+  keyring_controller_->GetDefaultKeyringInfo(base::BindOnce(
+      [](content::RenderFrameHost* rfh, GetAllowedAccountsCallback callback,
+         brave_wallet::mojom::KeyringInfoPtr keyring_info) {
+        std::vector<std::string> addresses;
+        for (const auto& account_info : keyring_info->account_infos) {
+          addresses.push_back(account_info->address);
+        }
+        permissions::BraveEthereumPermissionContext::GetAllowedAccounts(
+            rfh, addresses,
```

Comment:
This is not safe. You should pass the routing id and get the RFH in the callacbk

Reply:
Thanks, addressed in https://github.com/brave/brave-core/pull/9752/commits/5adf0beaaace95b3c324e4b6f0678532c84593d7.